### PR TITLE
Modify alert definition, scrape interval, config type

### PR DESCRIPTION
**Pull request by mithunmgore**
crc error alert definition modified to notify absent alerts and
to get increased value by 100 in 30 min.
Scrape interval is reduced since cacheing is happening.
Config is changed from yaml to json

Signed-off-by: Mithun Gore <mithungore@gmail.com>

### CODE REVIEW

File: prometheus-exporters/ucs-exporter/alerts/ucsmcrcerrors.alerts [hunk 1]
```diff
@@ -2,13 +2,15 @@ groups:
 - name: ucsmcrcerrors.alerts
   rules:
   - alert: UcsmCRCError
-    expr: max_over_time(ucsm_crc_error[20m]) > 0
+    expr: increase(ucsm_crc_error[30m]) > 100
```
Comment (christopherhans) [inline prometheus-exporters/ucs-exporter/alerts/ucsmcrcerrors.alerts:5]: `increase` should only be used with counter metrics (check: https://prometheus.io/docs/prometheus/latest/querying/functions/#increase)

The needed function would be `delta`

File: prometheus-exporters/ucs-exporter/alerts/ucsmcrcerrors.alerts [hunk 2]
```diff
@@ -2,13 +2,15 @@ groups:
...
+    for: 20m
     labels:
       severity: info
       tier: vmware
       service: compute
       context: "ucs-exporter"
       meta: "Server {{ $labels.server }} has crc errors for port {{ $labels.port }}"
+      no_alert_on_absence: "true"
```
Comment (christopherhans) [inline prometheus-exporters/ucs-exporter/alerts/ucsmcrcerrors.alerts:13]: While this disabled the absent alerts, it does not solve the root cause. I would suggest lowering the scrape interval and make sure that there is always a metric in the alert expression range available.

File: system/infra-monitoring/values.yaml [hunk 3]
```diff
@@ -665,7 +665,7 @@ ask1k_tests:
 
 ucs_exporter:
   enabled: false
-  scrapeInterval: 15m
+  scrapeInterval: 10m
```
Comment (christopherhans) [inline system/infra-monitoring/values.yaml:668]: Good.
See comment above.

File: prometheus-exporters/ucs-exporter/alerts/ucsmcrcerrors.alerts (hunk 1 above)
Comment (christopherhans) [reply prometheus-exporters/ucs-exporter/alerts/ucsmcrcerrors.alerts:5]: If you change the function to `delta`, we can close this PR: https://github.com/sapcc/helm-charts/pull/2306

**Review by christopherhans — CHANGES_REQUESTED** [review]

**Review by christopherhans — APPROVED** [review]
lgtm

File: prometheus-exporters/ucs-exporter/templates/ucs-config.yaml [hunk 4]
```diff
@@ -3,12 +3,15 @@ kind: ConfigMap
...
-  config.yaml: |
-    netbox:
-        netbox_url: {{ .Values.ucs.netbox_url }}
-        netbox_token: {{ .Values.ucs.netbox_token }}
-        ucs_hostname_format: {{ .Values.ucs.ucs_hostname_format }}
-        regions: [{{ .Values.global.region }}]
-        tenant: {{ .Values.ucs.tenant }}
-        query: {{ .Values.ucs.query }}
-        status: {{ .Values.ucs.status }}
+  config.json: |
```
Comment (viennaa) [inline prometheus-exporters/ucs-exporter/templates/ucs-config.yaml:6]: Why an ugly json?

File: prometheus-exporters/ucs-exporter/alerts/ucsmcrcerrors.alerts (hunk 2 above)
Comment (viennaa) [reply prometheus-exporters/ucs-exporter/alerts/ucsmcrcerrors.alerts:13]: Please remove, the absent alerts are actually pretty valuable.

**Review by viennaa — CHANGES_REQUESTED** [review]

File: prometheus-exporters/ucs-exporter/templates/ucs-config.yaml (hunk 4 above)
Comment (mithunmgore) [reply prometheus-exporters/ucs-exporter/templates/ucs-config.yaml:6]: I will take care of this in next deployment. I kept it as it is from Daniels pr.

Comment (viennaa) [reply prometheus-exporters/ucs-exporter/templates/ucs-config.yaml:6]: Is it an impediment which we have in code?

Comment (mithunmgore) [reply prometheus-exporters/ucs-exporter/templates/ucs-config.yaml:6]: Yes, we are using json package in code

Comment (mithunmgore) [reply prometheus-exporters/ucs-exporter/templates/ucs-config.yaml]: @viennaa please check now. updated to yaml

**Review by viennaa — APPROVED** [review]

**Pull request merged**